include/mapnik/enumeration.hpp

```diff
@@ -176,9 +177,11 @@ class MAPNIK_DECL enumeration {
      * */
     void from_string(std::string const& str)
     {
+        std::string str_copy(str);
+        std::replace(str_copy.begin(), str_copy.end(), '_', '-');
         for (unsigned i = 0; i < THE_MAX; ++i)
```

Comment:
^^ this is for back compatibility? If so then could you add a note about removal at Mapnik 4.x?